LayoutKit/Views/ReloadableViewUpdateManager.swift

```diff
@@ -0,0 +1,138 @@
+// Copyright 2016 LinkedIn Corp.
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+import Foundation
+
+/// Delegate for a `ReloadableViewUpdateManager`.
+protocol ReloadableViewUpdateManagerDelegate: class {
+    weak var reloadableView: ReloadableView? { get }
+    var currentArrangement: [Section<[LayoutArrangement]>] { get set }
+}
+
+/// Manages updating a `ReloadableView` and its backing data.
+/// Updates may be applied incrementally, or in a batch.
+class ReloadableViewUpdateManager {
+    private weak var delegate: ReloadableViewUpdateManagerDelegate?
+    private weak var operation: NSOperation?
+
+    static func make(delegate delegate: ReloadableViewUpdateManagerDelegate,
+                              operation: NSOperation,
+                              incremental: Bool) -> ReloadableViewUpdateManager {
+        return incremental ?
+            IncrementalUpdateManager(delegate: delegate, operation: operation) :
+            BatchUpdateManager(delegate: delegate, operation: operation)
+    }
+
+    private init(delegate: ReloadableViewUpdateManagerDelegate, operation: NSOperation) {
+        self.delegate = delegate
+        self.operation = operation
+    }
+
+    func apply(partialArrangement arrangement: [Section<[LayoutArrangement]>], insertedIndexPath: NSIndexPath) {
+        // implemented by subclass
```

Comment:
yeah was supposed to be an abstract class.

After taking a second look, I am going to rework this to use protocols for more clarity.